pkg/drivers/libhvee/libhvee_windows.go

```diff
@@ -0,0 +1,309 @@
+		if err != nil {
+			return err
+		}
+
+		err = vm.UpdateProcessorMemSettings(
+			func(ps *hypervctl.ProcessorSettings) {
+				ps.VirtualQuantity = uint64(update.CPU)
+			},
+			func(ms *hypervctl.MemorySettings) {})
+
+		if err != nil {
+			return err
+		}
+	}
+	if update.DiskCapacity != d.DiskCapacity {
+		if err := d.resizeDisk(int64(update.DiskCapacity)); err != nil {
+			log.Warnf("Machine: libhvee -> Failed to set disk size to %d", update.DiskCapacity)
+			return err
+		}
+	}
+	*d = update
+	return nil
+}
+
+// DriverName returns the name of the driver
+func (d *Driver) DriverName() string {
+	return "libhvee"
+}
+
+func (d *Driver) GetState() (state.State, error) {
+	log.Debugf("Machine: libhvee -> state")
+	vm, err := d.getMachine()
+	if err != nil {
+		return state.Error, err
+	}
+
+	log.Debugf("Machine: libhvee -> state: get")
+	vmState := vm.State()
+	switch vmState {
+	case hypervctl.Enabled:
+		log.Debugf("Machine: libhvee -> state: running")
+		return state.Running, nil
+	case hypervctl.Disabled:
+		log.Debugf("Machine: libhvee -> state: stopped")
+		return state.Stopped, nil
+	}
+
+	log.Debugf("Machine: libhvee -> state: unknown")
+	return state.Error, fmt.Errorf("Unknown state")
+}
+
+// PreCreateCheck checks that the machine creation process can be started safely.
+func (d *Driver) PreCreateCheck() error {
+	// Check that powershell was found
+	if _, err := exec.LookPath("powershell.exe"); err != nil {
+		return ErrPowerShellNotFound
+	}
+
+	// Check that hyperv is installed
+	if err := hypervAvailable(); err != nil {
+		return err
+	}
+
+	// Check that the user is an Administrator
+	isAdmin, err := isAdministrator()
+	if err != nil {
+		return err
+	}
+	if !isAdmin {
+		return ErrNotAdministrator
+	}
+
+	return nil
+}
+
+func (d *Driver) Create() error {
+	log.Debugf("Machine: libhvee -> creating: system settings")
+	systemSettings, err := hypervctl.NewSystemSettingsBuilder().
+		PrepareSystemSettings(d.MachineName, nil).
+		PrepareMemorySettings(
+			func(ms *hypervctl.MemorySettings) {
+				ms.DynamicMemoryEnabled = d.DynamicMemory
+				ms.VirtualQuantity = uint64(d.Memory)
+				ms.Reservation = 1024
+				ms.Limit = uint64(d.Memory)
+			}).
+		PrepareProcessorSettings(
+			func(ps *hypervctl.ProcessorSettings) {
+				ps.VirtualQuantity = uint64(d.CPU)
+			}).
+		Build()
+
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: copy disk image")
+	diskPath := d.getDiskPath()
+	if err := crcos.CopyFile(d.ImageSourcePath, diskPath); err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: hardware setup")
+	err = hypervctl.NewDriveSettingsBuilder(systemSettings).
+		AddScsiController().
+		AddSyntheticDiskDrive(0).
+		DefineVirtualHardDisk(diskPath,
+			func(vhdss *hypervctl.VirtualHardDiskStorageSettings) {}).
+		Finish().
+		Finish().
+		Finish().
+		Complete()
+
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: done")
+
+	return nil
```

Comment:
This was `return d.resizeDisk()` in the hyperv driver. This used to be needed to get the disk size correct when doing 
```
crc delete
crc config set disk-size 50
crc start
```
Are you sure it's no longer needed?

Reply:
Will check. Might have been missed as resize scenario has mostly been untested.

Though, our testsuite passed here?

Reply:
I've tested this, and the `resizeDisk` call is required for initial disk resize to work.